includes/Order/Hooks.php

```diff
@@ -263,10 +263,19 @@ public function on_sub_order_change( $order_id, $old_status, $new_status, $order
         // return if any child order is not completed
         $all_complete = true;
 
+        // Exclude manual gateways from auto complete order status update for digital products.
+        $excluded_gateways = apply_filters( 'dokan_order_status_update_exclusion_payment_gateways', array( 'bacs', 'cheque', 'cod' ) );
+
         foreach ( $sub_orders as $sub_order ) {
+            // if the order is a downloadable and virtual product, then we need to set the status to complete
+            if ( $order->is_paid() && ! in_array( $order->get_payment_method(), $excluded_gateways, true ) && ! $sub_order->needs_processing() ) {
+                $sub_order->set_status( 'completed', __( 'Mark as completed when it has digital products.', 'dokan-lite' ) );
```

Comment:
Pls change the note to **Marked as completed because it contains digital products.**

Reply:
vai, order note updated.